### PR TITLE
Fix full-test with-snark bugs

### PR DESCRIPTION
**Pull request by bkase**
## Changes

I rolled back the:

```
[`Signer of signature | `Genesis]
```

change, and went back to global-signer-key. I increased the `CODA_PROPOSAL_INTERVAL` for full-test with snark as well.

I also fixed various other small issues and made a half-working parallel-scan-state visualizer (that still was helpful in debugging).

Note: full-test passed in the `test_snark` profile on my machine (inside a docker container on a VM in a macbook pro) with the timeouts included here: (15min for part one, and 5min for part two -- see full_test.ml).

## Test Plan

I successfully built with the `testnet` profile with the secret global private signer key (and the snark successfully was created at compile time). We still will want to check this in a devnet as well.

This obsoletes #1104

### CODE REVIEW

**Comment by bkase** [comment]
@nholland94 will commandeer this PR tomorrow as I'll be away

**Comment by bkase** [comment]
If CircleCI doesn't pass we should try increasing timeouts (I know they're already quite large)

**Comment by bkase** [comment]
Wow it seems my computer is _way_ slower than Circle! It seems to be passing in only 6minutes.

File: src/lib/consensus/proof_of_signature.ml [hunk 1]
```diff
@@ -207,18 +199,11 @@ module Make (Inputs : Inputs_intf) :
...
-        signature
-        (Public_key.var_of_t Global_public_key.t)
-        (transition |> Snark_transition.blockchain_state)
-    and previous_state_was_neg_one =
-      State_hash.equal_var previous_state_hash
-        (State_hash.var_of_t (Protocol_state.hash Protocol_state.negative_one))
-    in
-    Snark_params.Tick.Boolean.(
-      signature_verifies || previous_state_was_neg_one)
+    Blockchain_state.Signature.Checked.verifies
```
Comment (imeckler) [inline src/lib/consensus/proof_of_signature.ml]: the signature is always good I guess?

File: src/lib/parallel_scan/parallel_scan.mli [hunk 2]
```diff
@@ -25,6 +25,21 @@ module State : sig
 
   val copy : ('a, 'd) t -> ('a, 'd) t
 
+  val visualize :
+    ('a, 'd) t -> draw_a:('a -> string) -> draw_d:('d -> string) -> string
+  (** Visualize produces a tree in a way that's a bit buggy, but still was
+   * helpful for debugging. [visualize state ~draw_a:(Fn.const "A") ~draw_d:(Fn.const "D")]
+   * creates a tree that looks like this:
+   *
```
Comment (imeckler) [inline src/lib/parallel_scan/parallel_scan.mli]: nice

File: src/lib/proposer/proposer.ml [hunk 3]
```diff
@@ -259,11 +259,18 @@ module Make (Inputs : Inputs_intf) :
...
                   ignore
-                    (let%map protocol_state_proof =
+                    (let t0 = Time.now time_controller in
+                     let%map protocol_state_proof =
                        Prover.prove ~prev_state:previous_protocol_state
                          ~prev_state_proof:previous_protocol_state_proof
                          ~next_state:protocol_state internal_transition
                      in
+                     let span = Time.diff (Time.now time_controller) t0 in
+                     Logger.info logger
```
Comment (imeckler) [inline src/lib/proposer/proposer.ml]: did you want to leave this in or nah?

Comment (bkase) [reply src/lib/proposer/proposer.ml]: Yes, I thought it would be nice

**Pull request merged**